src/js/tracks/text-track-list.js

```diff
@@ -89,15 +89,22 @@ TextTrackList.prototype.addTrack_ = function(track) {
 };
 
 TextTrackList.prototype.removeTrack_ = function(rtrack) {
-  let result = null;
   let track;
 
   for (let i = 0, l = this.length; i < l; i++) {
-    track = this[i];
-    if (track === rtrack) {
-      this.tracks_.splice(i, 1);
-      break;
+    if (this[i] !== rtrack) {
+      continue;
     }
+
+    track = this[i];
+
+    this.tracks_.splice(i, 1);
+
+    break;
+  }
+
+  if (track) {
```

Comment:
Should this check for `!track`? Otherwise we'll never trigger `removetrack` if we actually found and removed a track?